miden-tx/tests/test_miden_faucet_contract.rs

```diff
@@ -0,0 +1,433 @@
+use miden_lib::{
+    assembler,
+    common::{self, prepare_transaction, run_tx},
+};
+
+use assembly::ast::{ModuleAst, ProgramAst};
+use common::NodeIndex;
+use crypto::{hash::Hasher, Felt, StarkField, ZERO};
+
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountVault},
+    assets::{Asset, FungibleAsset},
+    notes::{Note, NoteEnvelope, NoteMetadata, NoteOrigin, NoteStub, NoteVault},
+    BlockHeader, ChainMmr,
+};
+use mock::{
+    account::{mock_account_storage, MockAccountType},
+    constants::ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN,
+    notes::AssetPreservationStatus,
+    transaction::mock_inputs_with_existing,
+    utils::prepare_word,
+};
+
+use miden_tx::{DataStore, TransactionExecutor};
+use vm_processor::{crypto::Rpo256, Digest, MemAdviceProvider};
+
+#[derive(Clone)]
+pub enum DataStoreError {
+    AccountNotFound(AccountId),
+    NoteNotFound(u32, NodeIndex),
+}
+
+#[derive(Clone)]
+
+pub struct MockDataStore {
+    pub account: Account,
+    pub block_header: BlockHeader,
+    pub block_chain: ChainMmr,
+    pub notes: Vec<Note>,
+}
```

Comment:
This seems to be identical to the `MockDataStore` defined in `miden-tx/src/tests.rs`. Could we use that one instead of duplicating it here?

Reply:
yes